contracts/reverseRegistrar/L2ReverseRegistrar.sol

```diff
@@ -2,68 +2,42 @@ pragma solidity >=0.8.4;
 
 import "../registry/ENS.sol";
 import "./IL2ReverseRegistrar.sol";
+import "./SignatureReverseResolver.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
 import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
 import "../resolvers/profiles/ITextResolver.sol";
 import "../resolvers/profiles/INameResolver.sol";
 import "../root/Controllable.sol";
 import "../resolvers/Multicallable.sol";
+import "../utils/LowLevelCallUtils.sol";
 
-error InvalidSignature();
-error SignatureOutOfDate();
-error Unauthorised();
 error NotOwnerOfContract();
 
-// @note Inception date
-// The inception date is in milliseconds, and so will be divided by 1000
-// when comparing to block.timestamp. This means that the date will be
-// rounded down to the nearest second.
-
+/**
+ * A L2 reverser registrar. Deployed to each L2 chain.
+ * The contract will be verified on L1 Reverse Resolver under the namespace specified at constructor
+ */
 contract L2ReverseRegistrar is
     Multicallable,
     Ownable,
-    ITextResolver,
-    INameResolver,
-    IL2ReverseRegistrar
+    IL2ReverseRegistrar,
+    SignatureReverseResolver
 {
     using ECDSA for bytes32;
-    mapping(bytes32 => uint256) public lastUpdated;
-    mapping(uint64 => mapping(bytes32 => mapping(string => string))) versionable_texts;
-    mapping(uint64 => mapping(bytes32 => string)) versionable_names;
-    mapping(bytes32 => uint64) internal recordVersions;
-    event VersionChanged(bytes32 indexed node, uint64 newVersion);
-    event ReverseClaimed(address indexed addr, bytes32 indexed node);
 
     bytes32 public immutable L2ReverseNode;
-    uint256 public immutable coinType;
-
-    // This is the hex encoding of the string 'abcdefghijklmnopqrstuvwxyz'
-    // It is used as a constant to lookup the characters of the hex address
-    bytes32 constant lookup =
-        0x3031323334353637383961626364656600000000000000000000000000000000;
 
-    /**
+    /*
      * @dev Constructor
+     * @param _L2ReverseNode The namespace to set. The converntion is '${coinType}.reverse'
+     * @param _chainId The chainId converted from the chainId of the chain this contract is deployed to.
      */
-    constructor(bytes32 _L2ReverseNode, uint256 _coinType) {
+    constructor(
+        bytes32 _L2ReverseNode,
+        uint256 _chainId
```

Comment:
I changed the constructor to pass chainId instead of cointype reflecting [your comment of another PR](https://github.com/ensdomains/evmgateway/pull/34/files#r1509124561). However, it seems a bit odd to pass chainId while reversenode still uses the coinType instead of chain Id [according to the ENSIP ](https://github.com/ensdomains/docs/pull/157/files#diff-25328e8b031181b77dc4241237ec9b26de59260f7c643435c1b632f0710d50b4R30). Should I reverse the change to use coinType instead?